src/services/partner.ts

```diff
@@ -0,0 +1,53 @@
+import { CallOverrides } from "@ethersproject/contracts";
+
+import { ChainId } from "../chain";
+import { ContractAddressId, ContractService, WrappedContract } from "../common";
+import { Context } from "../context";
+import { AddressProvider } from "./addressProvider";
+
+const YVECRV = "0xc5bDdf9843308380375a611c18B50Fb9341f502A";
+const PSLPYVBOOSTETH = "0xCeD67a187b923F0E5ebcc77C7f2F7da20099e378";
+
+/**
+ * [[PartnerService]] provides access to yearns partner contract.
+ * It's implemented in the form of a contract that lives on all networks
+ * supported by yearn.
+ */
+export class PartnerService<T extends ChainId> extends ContractService<T> {
+  static abi = ["function deposit(address vault, address partnerId, uint256 amount) external returns (uint256)"];
+  static IGNORED_ADDRESSES = [YVECRV, PSLPYVBOOSTETH];
+  partnerId: string;
+  static contractId = ContractAddressId.partner;
+
+  get contract(): Promise<WrappedContract> {
+    return this._getContract(PartnerService.abi, PartnerService.contractId, this.ctx);
+  }
+
+  get address(): Promise<string> {
+    return this.addressProvider.addressById(PartnerService.contractId);
```

Comment:
Think we need to handle the cases where this returns the 0 address now, e.g. for fantom or arbitrum. I think the deposit will always go to the partner contract now right?